agent/engine/docker_container_engine.go

```diff
@@ -848,3 +849,11 @@ func (dg *dockerGoClient) removeImage(imageName string) error {
 	}
 	return client.RemoveImage(imageName)
 }
+
+func (dg *dockerGoClient) LoadImage(opts docker.LoadImageOptions) error {
+	client, err := dg.dockerClient()
```

Comment:
The agent do have [timeout](https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/docker_container_engine.go#L46) for all the docker APIs, if the docker hangs or is very slow, it doesn't make sense to just wait.